client/lib/plans/index.js

```diff
@@ -66,6 +68,28 @@ export function getSitePlanSlug( siteID ) {
 	return get( site, 'plan.product_slug' );
 }
 
+function canUpgradeToPlan( planKey, site ) {
+	const offer = site.jetpack ? JETPACK : WPCOM;
```

Comment:
We should also be checking for if the site has an existing plan. We currently don't allow downgrades so for example if I had a site with a premium plan, I should never be able to checkout a Free or Premium plan.

If this is handled elsewhere, and you're checking the list of jetpack vs wpcom plans, I would suggest a different method name.